src/test/java/co/cask/http/HttpServerTest.java

```diff
@@ -385,8 +391,12 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     HttpUtil.setContentLength(request, 4);
     channel.writeAndFlush(request);
     HttpResponse response = queue.poll(10, TimeUnit.SECONDS);
-    Assert.assertEquals(200, response.status().code());
-    Assert.assertTrue(HttpUtil.isKeepAlive(response));
+    try {
+      Assert.assertEquals(200, response.status().code());
+      Assert.assertTrue(HttpUtil.isKeepAlive(response));
+    } finally {
+      ReferenceCountUtil.release(response);
```

Comment:
what is this doing? I don't see `retain` being called anywhere here...?

Reply:
It's because the `response` can be a `FullHttpResponse` object, which we retain it to the blocking queue.